### PR TITLE
EditorConfig + Contribution Guidelines

### PR DESCRIPTION
**Pull request by shivapoudel**
:gift:

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 
-# Use new container based environment
-sudo: false
-
 # Declare project language.
 # @link http://about.travis-ci.org/docs/user/languages/php/
 language: php
 
+# Use new container based environment
+sudo: false
```
Comment (GaryJones) [inline .travis.yml]: Why do it later? Logically we would want to know where we're going to run stuff, before knowing what we're going to run?

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+indent_style = tab
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.md]
+trim_trailing_whitespace = false
+indent_style = space
+indent_size = 2
+
+[*.txt]
```
Comment (GaryJones) [inline .editorconfig]: We don't have .txt or .json files, so these can be removed.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+trim_trailing_whitespace = false
+
+[*.json]
+insert_final_newline = false
+indent_style = space
+indent_size = 2
+
+[*.yml]
```
Comment (GaryJones) [inline .editorconfig]: Since this is the same as Markdown, you can change line 18 to `[{*.md,*.yml}]` and delete this block.

**Comment by GaryJones** [comment]
Please do these as separate branches / PRs, so if we need to revert one, we don't have to revert unrelated bits.

One for editorconfig, one for readme, etc.

And please squash the commits before pushing.

**Comment by shivapoudel** [comment]
Close in favor of #339 and #340

**Pull request closed without merging**